### ISSUE TITLE
sparse_categorical_crossentropy only works on single output

### ISSUE THREAD

**Issue by geovedi** [fchollet/keras#2792]
So I have this model that try to predict multiple outputs,

```python
    ...
    decoded_outputs = []
    for i, size in enumerate(target_sizes):
        dense = TimeDistributed(Dense(size, activation='softmax'), name='out_{0}'.format(i))
        decoded_outputs.append(dense(decoded))

    autoencoder = Model(input=[encoded_input], output=decoded_outputs, name='autoencoder')
    autoencoder.compile(loss='sparse_categorical_crossentropy', optimizer='adam')
```
but `standardize_input_data()` keeps complaining about the mismatch shape. 

As workaround, I have to use smaller `batch_size` (as my original idea was to reduce memory usage for train data) and `categorical_crossentropy`. But surely it would be great if `sparse_categorical_crossentropy` can handle multiple outputs.

**Comment by fchollet** [comment]
Did you try updating Keras? As specified in the checklist that you had to delete for posting this.

**Comment by geovedi** [comment]
> Did you try updating Keras?

yes, I have tested it with `1.0.3` (installed from `pip`) and also against the master branch.

this is the simplified model that I'm trying to have.. 

```
input_dim = 1
timestep = 10
latent_dim = 512
target_sizes = (300, 400)

input_ = Input(shape=(timestep, input_dim))
encoded = LSTM(latent_dim)(input_)

decoded = RepeatVector(timestep)(encoded)
decoded = LSTM(latent_dim, return_sequences=True)(decoded)
output_ = []
for _, size in enumerate(target_sizes):
    dense = TimeDistributed(Dense(size))
    output_.append(dense(decoded))

model = Model(input=[input_], output=output_)
model.compile(loss='sparse_categorical_crossentropy', optimizer='adam')

X_train = np.random.randint(target_sizes[1], size=(50, timestep, 1))
yy_train = np.random.randint(target_sizes[0], size=(50, timestep, 1))
zz_train = np.random.randint(target_sizes[1], size=(50, timestep, 1))

model.fit([X_train], [yy_train, zz_train], nb_epoch=10, batch_size=32, shuffle=True)
```

It will give raise the exception: `Exception: Error when checking model target: expected timedistributed_2 to have shape (None, 10, 400) but got array with shape (50, 10, 1)`

while this one is working.

```
output_2 = TimeDistributed(Dense(target_sizes[0]))(decoded)
model_2 = Model(input=input_, output=output_2)
model_2.compile(loss='sparse_categorical_crossentropy', optimizer='adam')

model_2.fit(X_train, yy_train, nb_epoch=10, batch_size=32, shuffle=True)
```

**Comment by mbollmann** [comment]
I only found out after 20 minutes of debugging, so I'll leave this here:

This issue has been fixed in commit 25c10af59694c6a61778f9111e3ca97dfd8971b4.

**Comment by geovedi** [comment]
thanks @mbollmann!

**Comment by RudrakshTuwani** [comment]
Hi ,

I seem to be getting the same error. I'm using Keras version 1.2 with Tensorflow backend. Here's my code. 

`
print('Build model...')
model = Sequential()
model.add(Embedding(10000, 100, dropout=0.2, input_length=num_steps))
model.add(LSTM(50, return_sequences=True))
model.add(Dropout(0.2))
model.add(TimeDistributed(Dense(10000, activation='softmax')))
model.compile(loss='sparse_categorical_crossentropy', optimizer='rmsprop', metrics=['accuracy'])

print(model.summary())
`

**Comment by DingKe** [comment]
Update keras to the newest github version, it's fixed in this [commit](https://github.com/fchollet/keras/commit/21023f7f9cab8042f1e4359348e8416f410855e0).

**Comment by RudrakshTuwani** [comment]
Thanks a lot ! It works now.
